tsdb/index/tsi1/partition.go

```diff
@@ -348,22 +349,34 @@ func (p *Partition) buildSeriesSet() error {
 }
 
 // CurrentCompactionN returns the number of compactions currently running.
-func (p *Partition) CurrentCompactionN() int {
-	p.mu.RLock()
-	defer p.mu.RUnlock()
-	return p.currentCompactionN
+func (p *Partition) CurrentCompactionN() int32 {
+	return p.currentCompactionN.Load()
 }
 
 // Wait will block until all compactions are finished.
 // Must only be called while they are disabled.
 func (p *Partition) Wait() {
 	ticker := time.NewTicker(10 * time.Millisecond)
 	defer ticker.Stop()
+
+	// Debug level timeout
+	timeoutDuration := 24 * time.Hour
+
 	for {
 		if p.CurrentCompactionN() == 0 {
 			return
 		}
-		<-ticker.C
+		select {
+		case <-time.After(timeoutDuration):
```

Comment:
`time.After` creates a new channel every call. You need to move it outside the loop.  See here: https://stackoverflow.com/a/39212626

This only prints the warning once., though.  If you want to print the warning repeatedly (every `timeoutDuration`) then you should use an `if` in the ticker case and check elapsed time.